packages/i18next/src/lib/InternationalizationHandler.ts

```diff
@@ -0,0 +1,203 @@
+import { getRootData } from '@sapphire/pieces';
+import { isFunction, NonNullObject } from '@sapphire/utilities';
+import { opendir } from 'fs/promises';
+import i18next, { StringMap, TFunction, TFunctionKeys, TFunctionResult, TOptions } from 'i18next';
+import Backend, { i18nextFsBackend } from 'i18next-fs-backend';
+import { join } from 'path';
+import type { InternationalizationContext, InternationalizationOptions } from './types';
+
+/**
+ * A generalized class for handling `i18next` JSON files and their discovery.
+ * @since 1.0.0
+ */
+export class InternationalizationHandler {
```

Comment:
It's odd to use abbreviations in class names, and even odder to use numbers in them.

Reply:
> It's odd to use abbreviations in class names, and even odder to use numbers in them.

Is "It's odd" really the best reasoning to use?

1. Abbreviation is less verbose and less typographical, leaning to functionality and meaning (as it should)
2. While potenitally less readable at first, the term is common enough that putting i18n or l10n anywhere resolves to internationalization and localization, making it perfectly understandable and unambiguous

Other languages have things like `BTree` which is a binary tree, `ptr` which is pointer, et cetera. Hell, JS / TS even have `const` as short for constant. What's the difference in abbreviating a class name, aside from making the code neater and less of a pain to work with?

Reply:
- Not everyone may know what `i18n` stands for (think new people in the coding world)
- I prefer classes and such be verbose not shorthanded whenever possible, to be crystal clear whats what

Reply:
Vlad, I addressed your first point pre-emptively in my second one, and the latter is a matter of preference, but as far as "Issues of Typography" goes, things should lean towards being concise but meaningful rather than overly verbose when it isn't necessary.

In any case, you guys are the lead developers, I just thought I'd drop my two cents in.